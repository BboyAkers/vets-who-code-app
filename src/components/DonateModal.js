import React from 'react';
import Modal from 'react-modal';

Modal.setAppElement('#___gatsby');

const container = {
  content: {
    top: '20%',
    bottom: '20%',
    height: 'auto'
  }
};

const closeButton = {
  right: '8px'
};

export default function DonateModal({ modalIsOpen, openModal, closeModal }) {
  return (
    <Modal
      style={container}
      contentLabel="Donate"
      isOpen={modalIsOpen}
      onRequestClose={closeModal}
      shouldCloseOnOverlayClick={true}
    >
      <iframe
        src="https://donorbox.org/embed/vetswhocode-donation?show_content=true"
        style={{
          maxWidth: '100%',
          minWidth: '100%',
          maxHeight: 'none!important'
        }}
        seamless="seamless"
        name="donorbox"
        scrolling="no"
        allowpaymentrequest
        width="100%"
        height="685px"
        frameBorder={0}
      />
    </Modal>
  );
}
