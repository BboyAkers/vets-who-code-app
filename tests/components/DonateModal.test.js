import React from 'react'
import DonateModal from '../../src/components/DonateModal'

describe('<DonateModal />', () => {
  let wrapper

  beforeEach(() => wrapper = shallow(<DonateModal />))

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})