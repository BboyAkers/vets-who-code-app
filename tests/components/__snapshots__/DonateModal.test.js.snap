// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DonateModal /> should render correctly 1`] = `
<Modal
  contentLabel="Donate"
  shouldCloseOnOverlayClick={true}
  style={
    Object {
      "content": Object {
        "bottom": "20%",
        "height": "auto",
        "top": "20%",
      },
    }
  }
>
  <iframe
    allowpaymentrequest={true}
    frameBorder={0}
    height="685px"
    name="donorbox"
    scrolling="no"
    seamless="seamless"
    src="https://donorbox.org/embed/vetswhocode-donation?show_content=true"
    style={
      Object {
        "maxHeight": "none!important",
        "maxWidth": "100%",
        "minWidth": "100%",
      }
    }
    width="100%"
  />
</Modal>
`;
